src/DynamoCore/Interfaces/IWorkspaceModel.cs

```diff
@@ -21,5 +21,11 @@ public interface IWorkspaceModel
         event Action NodesCleared;
         event Action<ConnectorModel> ConnectorAdded;
         event Action<ConnectorModel> ConnectorDeleted;
+
+        /// <summary>
+        /// Implement to record node modification for undo/redo
+        /// </summary>
+        /// <param name="nodeModel"></param>
+        void RecordModelForModification(ModelBase nodeModel);
```

Comment:
On a second thought, please make this take `List<ModelBase> models` so that multiple modifications can be recorded in one action group through the interface. For your case, just send down a `new [] { model }` will do.